docs/en/03_Upgrading/index.md

```diff
@@ -37,123 +89,1015 @@ and also checking the `composer.json` files in [recipe-core](https://github.com/
+
+The equivalent `.env` file will look like this.
+```bash
+## Environment
+SS_ENVIRONMENT_TYPE="dev"
+SS_DEFAULT_ADMIN_USERNAME="admin"
+SS_DEFAULT_ADMIN_PASSWORD="password"
+SS_BASE_URL="http://localhost/"
+
+## Database
+SS_DATABASE_CHOOSE_NAME="true"
+SS_DATABASE_CLASS="MySQLDatabase"
+SS_DATABASE_USERNAME="root"
+SS_DATABASE_PASSWORD=""
+SS_DATABASE_SERVER="127.0.0.1"
 ```
 
-### Switching to new `app/src` structure
+### Cleaning up `mysite/_config.php` after your environment configuration upgrade
+
+You'll need to clean up your `mysite/_config.php` file after upgrading your environment file.
+
+The global values `$database` and `$databaseConfig` have been deprecated. Your database configuration details should be stored in your `.env` file. If you want to keep your database configuration in `_config.php`, you can use the new `DB::setConfig()` api, however this is discouraged.
+
+Requiring `conf/ConfigureFromEnv.php` is is no longer necessary. You should remove any references to it in `_config.php`.
+
+The removal of the `_ss_environment.php` file means that conditional logic is no longer available in the environment
+variable set-up process. This encouraged bad practice and should be avoided. If you still require conditional logic early in the bootstrap, this is best placed in the `_config.php` files.
+
+To access environment variables, use the `SilverStripe\Core\Environment::getEnv()` method. To define environment variables, use the `SilverStripe\Core\Environment::setEnv()` method.
+
+### Finalising your environment upgrade
+It's inadvisable to track your `.env` file in your source control system as it might contain sensitive information.
+
+You should ignore the `.env` file by adding an entry to your `.gitignore` file. You can create a sample environment configuration by duplicating your `.env` file as `.env.sample`, and removing sensitive information from it.
+
+You can safely delete your legacy `_ss_environement.php` if you want.
+
+This is a good point to commit your changes to your source control system before moving on to the next step.  
+
+## Step 3 - Namespacing your project (optional) {#step3}
+
+Namespacing your code is an optional step. It is recommended and will help future-proof your code base.
+
+To learn more about PHP namespace:
+* Read the [official Namespace PHP documentation](http://php.net/manual/en/language.namespaces.php)
+* Read the [PSR-4: Autoloader standard](https://www.php-fig.org/psr/psr-4/).
+
+### Before you start namespacing your codebase
+
+You need to choose a root namespace for your project. We recommend following the `Vendor\Package` pattern.
+
+The `Page` and `PageController` classes *must* be defined in the global namespace (or without a namespace).
+
+If you want your codebase to comply with the PSR-4 standard, make sure sub-directories of of your code folder are using the _upper case camel_ naming convention. For example, `mysite/code/page_types` should be renamed to `mysite/code/PageTypes`.
```

Comment:
Upper Case Camel it is. 

I would avoid describing this as _title case_ to avoid confusion with the [heading writing style](http://www.grammar-monster.com/glossary/title_case.htm).